### ISSUE TITLE
[1.17.0-RC2] Incorrect ROOT_URL  warning when using proxy through apache

### ISSUE THREAD

**Issue by gd197** [go-gitea/gitea#20501]
### Description

Hello,
my configuration : 
app.ini
`ROOT_URL                        = https://host.my.company.corp/gitea`
httpd gitea.conf

ProxyPass         /gitea  http://localhost:8888 nocanon
ProxyPassReverse  /gitea  http://localhost:8888
ProxyRequests     Off
AllowEncodedSlashes NoDecode
ProxyPreserveHost On

output logs on starting gitea : 
`2022/07/27 10:34:14 cmd/web.go:221:listen() [I] [62e0f886-27] AppURL(ROOT_URL): https://host.my.company.corp/gitea/`

please note the trailing slash at the end of the ROOT_URL in the log
ROOT_URL warning occurs when trying to access to https://host.my.company.corp/gitea
ROOT_URL warning does not occurs when trying to access to https://host.my.company.corp/gitea/
Did not find a way to force apache to rewrite url before proxying to application listener

### Gitea Version

1.17.0-rc2

### Can you reproduce the bug on the Gitea demo site?

No

### Operating System

RHEL 7.9

### How are you running Gitea?

Using gitea build from github release tab

### Database

MySQL

**Comment by zeripath** [comment]
Please read the instructions at https://docs.gitea.io/en-us/reverse-proxies/#apache-httpd

Your problem is occurring because of the `ProxyPassReverse` directive and `ProxyPreserveHost`.

**Comment by wxiaoguang** [comment]
I'd like to keep it open and propose a more flexible fix for the URL detection.

**Comment by wxiaoguang** [comment]
The fix: Some users visit "https://domain/gitea" while appUrl is "https://domain/gitea/", there should be no warning
* #20502

**Comment by gd197** [comment]
removing ProxyPassReverse directive and ProxyPreserveHost but the issue remains the same.
Tried in app.ini ROOT_URL with and without trailing slash in this configuration does not change the behavior as well

**Comment by wxiaoguang** [comment]
The PR https://github.com/go-gitea/gitea/pull/20502 will fix the false positive warning.

At the moment, you could bypass it by rewrite (using `mod_rewrite`) the "/gitea" to "/gitea/" 😂

**Comment by gd197** [comment]
thank you very much for your super fast reply, turning around mod_rewrite since yesterday but still not able to make it works, my guess is that apache is proxying first

**Comment by wxiaoguang** [comment]
You need something like this to bypass:

```
    RewriteEngine on
    RewriteRule ^/gitea$ /gitea/ [R=302,L]
```

**Comment by gd197** [comment]
this is what I'm trying but no luck :( 
mod_rewrite is there and enabled 
Anyway, your fix will solve that issue I can wait for.
Thanks again for your work and tips

**Comment by wxiaoguang** [comment]
Now the fix has been released to latest 1.17. You can get it from https://dl.gitea.io/gitea/1.17 or docker  gitea:1.17-dev


ps: I have tested the Apache config, it should work (the rewirte need R=302 to do the redirection)

```
<VirtualHost *:8001>
    RewriteEngine on
    RewriteRule ^/gitea$ /gitea/ [R=302,L]

    DocumentRoot "/var/www/html/8001"
    <Proxy *>
         Order allow,deny
         Allow from all
    </Proxy>
    AllowEncodedSlashes NoDecode
    ProxyPass /gitea http://localhost:8002 nocanon
</VirtualHost>

<VirtualHost *:8002>
    DocumentRoot "/var/www/html/8002"
</VirtualHost>
```

```
$ curl -v 127.0.0.1:8001/gitea
> GET /gitea HTTP/1.1
> Host: 127.0.0.1:8001
>
< HTTP/1.1 302 Found
< Location: http://127.0.0.1:8001/gitea/
< Content-Length: 289
< ....
<p>The document has moved <a href="http://127.0.0.1:8001/gitea/">here</a>.</p>
<address>Apache/2.4.54 (Debian) Server at 127.0.0.1 Port 8001</address>
```
